content/news/2023-03-04-bevy-0.10/index.md

```diff
@@ -0,0 +1,1546 @@
+
+In [#6681](https://github.com/bevyengine/bevy/pull/6681), by tightly packing entity location metadata and avoiding extra memory lookups, we've significantly reduced the overhead when making random query lookups via `Query::get`, seeing up to a 43% reduction in the overhead spent in `Query::get` and `World::get`.
+
+In [#6800](https://github.com/bevyengine/bevy/pull/6800) and [#6902](https://github.com/bevyengine/bevy/pull/6902), we've found that rustc can optimize out compile-time constant branches across function boundaries, moving the branch from runtime to compile time, has resulted in up to a 50% reduction in overhead when using `EntityRef::get`, `EntityMut::insert`, `EntityMut::remove`, and their variants.
+
+In [#6391](https://github.com/bevyengine/bevy/pull/6391), we've reworked `CommandQueue`'s internals to be more CPU-cache friendly, which has shown up to a 37% speedup when encoding and applying commands.
+
+## Adaptive Batching for Parallel Query Iteration
+
+<div class="release-feature-authors">authors: @james7132</div>
+
+`Query::par_for_each` has been the tool everyone reaches for when their queries get too big to run single-threaded. Got 10,0000 entities running around on your screen? No problem, `Query::par_for_each` chunks it up into smaller batches and distributes the workload over multiple threads. However, in **Bevy 0.9** and before, `Query::par_for_each` required callers to provide a batch size to help tune these batches for maximum performance. This rather opaque knob often resulted in users just randomly picking a value and rolling with it, or fine tuning the value based on their development machines. Unfortunately, the most effective value is dependent on the runtime environment (i.e. how many logical cores does a player's computer have) and the state of the ECS World (i.e. how many entities are matched?). Ultimately most users of the API just chose a flat number and lived with the results, good or bad.
+
+```rust
+// 0.9
+const QUERY_BATCH_SIZE: usize = 32;
+
+query.par_for_each(QUERY_BATCH_SIZE, |mut component| {
+   // ...
+});
+```
+
+In 0.10, you no longer need to provide a batch size! Bevy will automatically evaluate the state of the World and task pools and select a batch size using a heuristic to ensure sufficient parallelism, without incurring too much overhead. This makes parallel queries as easy to use as normal single-threaded queries! While great for most typical use cases, these heuristics may not be suitable for every workload, so we've provided an escape hatch for those who need finer control over the workload distribution. In the future, we may further tune the backing heuristics to try to get the default to be closer to optimal in these workloads. For more complete details on how the heuristics work and what workloads may be considered atypical, see the documentation of [`Query::par_iter`].
```

Comment:
This mentions an escape hatch and documentation about heuristics, but I am not seeing any of that at http://dev-docs.bevyengine.org/bevy/ecs/system/struct.Query.html#method.par_iter

Not sure if that's the wrong thing to link to, or if the docs don't exist. The closest I could find was [BatchingStrategy](http://dev-docs.bevyengine.org/bevy/ecs/query/struct.BatchingStrategy.html#).

Reply:
While poking around, I discovered/fixed https://github.com/bevyengine/bevy/pull/7900, but the currently link still doesn't seem to get me to the information described.

Reply:
Added BatchingStrategy info and removed the incorrect sentence about the docs.